From e4aedc69e341c1e2cf49c5dbaaef0c9e4ee95102 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Fri, 30 Oct 2026 16:36:55 +0100
Subject: [PATCH] Mark out-of-process applets as unsupported on Wayland

Out-of-process applets rely on GtkPlug, which is not supported on
Wayland. Ignore the Wayland platform declaration for these applets
when determining platform compatibility.
---
 .../panel-applets-manager-dbus.c                           | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/mate-panel/libmate-panel-applet-private/panel-applets-manager-dbus.c b/mate-panel/libmate-panel-applet-private/panel-applets-manager-dbus.c
index d78742e3a..c71e7091a 100644
--- a/mate-panel/libmate-panel-applet-private/panel-applets-manager-dbus.c
+++ b/mate-panel/libmate-panel-applet-private/panel-applets-manager-dbus.c
@@ -108,6 +108,7 @@ _mate_panel_applets_manager_get_applet_info (GKeyFile    *applet_file,
 	char            **supported_platforms;
 	gboolean          x11_supported;
 	gboolean          wayland_supported;
+	gboolean          in_process;
 
 	iid = g_strdup_printf ("%s::%s", factory_id, group);
 	name = g_key_file_get_locale_string (applet_file, group,
@@ -144,6 +145,12 @@ _mate_panel_applets_manager_get_applet_info (GKeyFile    *applet_file,
 		}
 	}
 
+	/* Out-of-process applets require GtkPlug, which is not supported on Wayland */
+	in_process = g_key_file_get_boolean (applet_file, "Applet Factory",
+					     "InProcess", NULL);
+	if (!in_process)
+		wayland_supported = FALSE;
+
 	info = mate_panel_applet_info_new (iid, name, comment, icon, (const char **) old_ids,
 					   x11_supported, wayland_supported);
 
